include/envoy/common/optional.h

```diff
@@ -1,6 +1,7 @@
 #pragma once
 
 #include "envoy/common/exception.h"
+#include "absl/types/optional.h"
```

Comment:
I rather folks just consume this directly across the tree, rather than having indirection here via our own wrapper header/build target.

Reply:
ack

Reply:
Generally I think it's just clearer to use the actual direct includes. I think we'll eventually go with the C++17 `std::optional,` which `absl::optional` is supposed to be compatible with, so we'll just do a giant `sed` at that point.